provider/pkg/await/awaiters.go

```diff
@@ -613,6 +615,42 @@ func untilCoreV1ResourceQuotaUpdated(c updateAwaitConfig) error {
 
 // --------------------------------------------------------------------------
 
+// core/v1/Secret
+
+// --------------------------------------------------------------------------
+
+func untilCoreV1SecretInitialized(c createAwaitConfig) error {
+	//
+	// Some types secrets do not have data available imediately and therefore are not considered initialized where data map is empty.
+	//
+	secretType, _ := openapi.Pluck(c.currentInputs.Object, "type")
+
+	// Other secret types are not generated by controller therefore we do not need to create a watcher for them.
+	if secretType != "kubernetes.io/service-account-token" {
+		return nil
+	}
+
+	secretDataAllocated := func(secret *unstructured.Unstructured) bool {
+		data, _ := openapi.Pluck(secret.Object, "data")
+		if secretData, isMap := data.(map[string][]byte); isMap {
+			// We don't need to wait for any specific initialization. Most of the time we create a secret with
+			// empty data which are propagated by controller so it's enough to check if data map is not empty.
+			return len(secretData) > 0
+		}
+		return false
+	}
+
+	client, err := c.clientSet.ResourceClient(c.currentOutputs.GroupVersionKind(), c.currentOutputs.GetNamespace())
+	if err != nil {
+		return err
+	}
+
+	return watcher.ForObject(c.ctx, client, c.currentOutputs.GetName()).
+		WatchUntil(secretDataAllocated, 5*time.Second)
```

Comment:
I wonder if the timeout should be configurable. I'm assuming that this operation is normally pretty quick. so it might be ok.

Reply:
hmmm, maybe it could. But yes, the operation is quick.  I always just read the secret immediately after creation without any timeout. I did not find it exampled in docs so I tried to dig it from code and looks like there is just handler for that secret type: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/serviceaccount/tokens_controller.go#L114

Reply:
I'm ok with this for now. We can adjust it later if users report problems with it timing out.